packages/main/src/sync.ts

```diff
@@ -25,23 +26,33 @@ const cachePolicy = {
 
 const BATCH_SIZE = 50
 
-export async function sync(accountId: cache.ID, manager: ConnectionManager) {
+async function boxSyncer(accountId: cache.ID, manager: ConnectionManager) {
+  let boxSync
+  let box
```

Comment:
It looks like you started something involving assigning promises to variables and ended up going in a different direction. These variables do not need to be declared outside of the loop.

Because `boxSyncer` is marked `async` it will return a promise regardless of whether there is a return statement. The promise that is returned will not resolve until the promises from all `await` expressions in the function resolve.